sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py

```diff
@@ -0,0 +1,284 @@
+                        trace.INVALID_SPAN, context=context
+                    )
+            elif stripped_kv_pair.startswith(self.SAMPLED_FLAG_KEY):
+                is_sampled_flag_valid = True
+
+                if len(value) != self.SAMPLED_FLAG_LENGTH:
+                    is_sampled_flag_valid = False
+
+                if is_sampled_flag_valid:
+                    sampled_flag = value[0]
+                    if sampled_flag == self.IS_SAMPLED:
+                        sampled = True
+                    elif sampled_flag == self.NOT_SAMPLED:
+                        sampled = False
+                    else:
+                        is_sampled_flag_valid = False
+
+                if not is_sampled_flag_valid:
+                    _logger.error(
+                        (
+                            "Invalid Sampling flag in X-Ray trace header: '%s' with value '%s'. Returning INVALID span context.",
+                            self.TRACE_HEADER_KEY,
+                            trace_header,
+                        )
+                    )
+                    return trace.set_span_in_context(
+                        trace.INVALID_SPAN, context=context
+                    )
+
+        options = 0
+        if sampled:
+            options |= trace.TraceFlags.SAMPLED
+
+        span_context = trace.SpanContext(
+            trace_id=trace_id,
+            span_id=span_id,
+            is_remote=True,
+            trace_flags=trace.TraceFlags(options),
+            trace_state=trace.TraceState(),
+        )
+
+        if not span_context.is_valid:
+            _logger.error(
+                "Invalid Span Extracted. Insertting INVALID span into provided context."
+            )
+            return trace.set_span_in_context(
+                trace.INVALID_SPAN, context=context
+            )
+
+        return trace.set_span_in_context(
+            trace.DefaultSpan(span_context), context=context
+        )
+
+    def inject(
+        self,
+        set_in_carrier: Setter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: typing.Optional[Context] = None,
+    ) -> None:
+        span = trace.get_current_span(context=context)
+
+        span_context = span.get_span_context()
+        if not span_context.is_valid:
+            return
+
+        otel_trace_id = "{:032x}".format(span_context.trace_id)
+        xray_trace_id = (
+            self.TRACE_ID_VERSION
+            + self.TRACE_ID_DELIMITER
+            + otel_trace_id[: self.TRACE_ID_FIRST_PART_LENGTH]
+            + self.TRACE_ID_DELIMITER
+            + otel_trace_id[self.TRACE_ID_FIRST_PART_LENGTH :]
+        )
+
+        parent_id = "{:016x}".format(span_context.span_id)
+
+        sampling_flag = (
+            self.IS_SAMPLED
+            if span_context.trace_flags & trace.TraceFlags.SAMPLED
+            else self.NOT_SAMPLED
+        )
+
+        # TODO: Add OT trace state to the X-Ray trace header
+
+        trace_header = (
```

Comment:
This is a great suggestion! Thank you!

Reply:
Just a note that we go with this style in Java since the performance difference vs a join is significant there. Maybe not a big deal in Python though.